squash-honeycomb.ipynb

```diff
@@ -0,0 +1,381 @@
+    "\n",
+    "libhoney.init(writekey=HONEY_API_KEY, dataset=\"single-event-demo\")\n",
+    "libhoney.send_now(data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "That's it! Check the new dataset at https://ui.honeycomb.io/lsst-square/datasets"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "NOTE: this demo dataset may already exist from previous executions of this notebook. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Now, let's send all squash data to Honeycomb\n",
+    "Actually, just the scalar metrics and some context information."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Create the `squash-demo` dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "libhoney.init(writekey=HONEY_API_KEY, dataset=\"squash-demo\")\n",
+    "builder = libhoney.Builder()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And loop over all jobs:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from datetime import datetime\n",
+    "    \n",
+    "jobs = requests.get(SQUASH_API_URL + \"/jobs\").json()\n",
+    "\n",
+    "for job_id in jobs['ids']:\n",
+    "\n",
+    "    r = requests.get(SQUASH_API_URL + \"/job/{}\".format(job_id)).json()\n",
+    "    \n",
+    "    if r['ci_dataset'] == 'unknown' or r['ci_dataset'] == 'decam':\n",
+    "        continue\n",
+    "    \n",
+    "    print('Sending event for job {}...'.format(job_id))\n",
+    "\n",
+    "    \n",
+    "    # Spawn a new event and override the timestamp\n",
+    "    event = builder.new_event()\n",
+    "    event.add_field('id', job_id)\n",
+    "    event.add_field('filter_name', r['meta']['filter_name'])\n",
+    "    event.add_field('dataset', r['ci_dataset'])\n",
+    "    \n",
+    "    for meas in r['measurements']:\n",
+    "        event.add_field(meas['metric'], meas['value'])\n",
+    "    \n",
+    "    event.created_at = datetime.strptime(r['date_created'], \"%Y-%m-%dT%H:%M:%SZ\")\n",
+    "    \n",
+    "    event.send()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Querying the dataset\n",
+    "\n",
+    "Querying a Honeycomb dataset in a particular way will produce a series of events (or a time series). The query builder is great tool for fast exploration from the Honeycomb UI. It is also possible to [specify queries programatically](https://docs.honeycomb.io/api/query-specification/)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The SQuaSH `validate_drp.AM1` metric on a given dataset and filter can be obtained obtained from this query:"
+   ]
+  },
+  {
```

Comment:
I appreciate being explicit about the syntax, but how do you execute this query and what do you get back.  It would be good to add that.

Reply:
Oh, sorry.  I should read :).  I see you use this query a couple of cells down.  Disregard this comment.